src/StreamWrapper/IncludeInterceptor.php

```diff
@@ -82,33 +87,43 @@ public static function enable(): void
         }
         stream_wrapper_unregister('file');
         stream_wrapper_register('file', __CLASS__);
+        self::$isEnabled = true;
     }
 
     public static function disable(): void
     {
         stream_wrapper_restore('file');
+        self::$isEnabled = false;
+    }
+
+    public static function isEnabled(): bool
+    {
+        return self::$isEnabled;
     }
 
     public function stream_open($path, $mode, $options)
     {
         self::disable();
+
         $including = (bool) ($options & self::STREAM_OPEN_FOR_INCLUDE);
 
-        if ($including) {
-            if ($path === self::$intercept || realpath($path) === self::$intercept) {
-                $this->fp = fopen(self::$replacement, 'r');
-                self::enable();
+        try {
+            if ($including) {
+                if ($path === self::$intercept || realpath($path) === self::$intercept) {
+                    $this->fp = fopen(self::$replacement, 'r');
 
-                return true;
+                    return true;
+                }
             }
-        }
 
-        if (isset($this->context)) {
-            $this->fp = fopen($path, $mode, (bool) $options, $this->context);
-        } else {
-            $this->fp = fopen($path, $mode, (bool) $options);
+            if (isset($this->context)) {
+                $this->fp = fopen($path, $mode, (bool) $options, $this->context);
+            } else {
+                $this->fp = fopen($path, $mode, (bool) $options);
```

Comment:
Not that it is in any way the point of this PR, but why can't this if/else branch be condensed down to just the following?

```php
$this->fp = fopen($path, $mode, (bool) $options, $this->context ?? null);
```

Reply:
it could even be `fopen($path, $mode, (bool) $options, $this->context)` all the time, as `$this->context` defaults to null

Reply:
@BackEndTea I hope that's fine by you, I applied this suggestion throughout the file as it makes the code a lot clearer to understand.